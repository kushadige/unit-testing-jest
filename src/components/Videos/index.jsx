import React from 'react'

const Videos = ({ videos = [] }) => {
  return videos.length === 0 ? (
        <h1>No videos found.</h1>
    ) : (
        <div>
            {videos.map((video, idx) => (
                <h3 key={idx}>{video}</h3>
            ))}
        </div>
    )
}

export default Videos