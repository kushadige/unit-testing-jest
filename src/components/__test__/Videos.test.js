import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Videos from '../Videos/index';

test('Videos component renders correctly', () => {
    const { container } = render(<Videos />);

    expect(container).toBeInTheDocument();
});

test('Videos runs correctly on empty state', () => {
    const tree = renderer.create(<Videos />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Videos runs correctly with 1 item', () => {
    const tree = renderer.create(<Videos videos={["Udemy"]} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Videos runs correctly with multiple item', () => {
    const tree = renderer.create(<Videos videos={["Udemy", "Youtube", "Vimeo", "Tutorial"]} />).toJSON();

    expect(tree).toMatchSnapshot(); 
});