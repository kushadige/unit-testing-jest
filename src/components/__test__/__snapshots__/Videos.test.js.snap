// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos runs correctly on empty state 1`] = `
<h1>
  No videos found.
</h1>
`;

exports[`Videos runs correctly with 1 item 1`] = `
<div>
  <h3>
    Udemy
  </h3>
</div>
`;

exports[`Videos runs correctly with multiple item 1`] = `
<div>
  <h3>
    Udemy
  </h3>
  <h3>
    Youtube
  </h3>
  <h3>
    Vimeo
  </h3>
  <h3>
    Tutorial
  </h3>
</div>
`;
